vagrant/provisioning/roles/pki-aws-client/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+- name: folder structure for app, data and logs
+  become: yes
+  file:
+    path: "{{ item }}"
+    state: directory
+  loop:
+    - "{{ root_folder }}"
+    - "{{ root_folder }}/app"
+    - "{{ root_folder }}/data"
+    - "{{ root_folder }}/log"
+    - "{{ root_folder }}/install"
+    - "{{ root_folder }}/common"
+    - "{{ root_folder }}/tmp"
+
+- name: install epel (needed first)
+  become: yes
+  yum:
+    state: installed
+    name:
+      - epel-release
+
+- name: install required packages
+  become: yes
+  yum:
+    state: installed
+    name:
+      - python-pip
+      - java-11-openjdk
+      - java-11-openjdk-devel
+
+- name: install Python crypto library
+  become: yes
+  pip:
+    name: cryptography
+    extra_args: --upgrade
+
+- name: pki facts
+  set_fact:
+    ssl_key: "/etc/pki/tls/private/{{ ansible_facts.hostname }}.pem"
+    ssl_cert: "/etc/pki/tls/certs/{{ ansible_facts.hostname }}.pem"
+    ssl_chain_cert: "/etc/pki/tls/certs/arkcase-chain.pem"
+    java_key_store: "{{ root_folder }}/common/arkcase.ks"
+    java_key_store_jks: "{{ root_folder }}/common/arkcase-jks.ks"
+    java_trust_store: "{{ root_folder }}/common/arkcase.ts"
+    java_key_store_pass: "{{ ansible_facts.fqdn }}"
+    java_trust_store_pass: "{{ ansible_facts.fqdn }}"
+    java_p12_store: "{{ root_folder }}/common/arkcase.p12"
+
+- name: ensure secure permissions on private keys
+  become: yes
+  file: 
+    path: "{{ ssl_key }}"
+    mode: 0600
+
+
+- name: generate the chain cert file
+  become: yes
+  shell: cat {{ ssl_intermediate_cert }} {{ ssl_ca }} > {{ ssl_chain_cert }}
+
+- name: Java keystore and trust store
```

Comment:
This and the next task are the same as the ones in pki_client. We should extract these in a common role. Maybe arkcase_pki role?

Reply:
I'm expecting to refactor later on, that is why the PR is in draft.  I'll try to restructure more cleanly.